src/Symfony/Component/DependencyInjection/Argument/ServiceClosureArgument.php

```diff
@@ -23,7 +23,7 @@ class ServiceClosureArgument implements ArgumentInterface
 {
     private array $values;
 
-    public function __construct(Reference $reference)
+    public function __construct(Reference $reference, private ?string $class = null)
```

Comment:
do we really need the argument?
can't we resolve the class when dumping the closure instead?
that'd allow not changing any loaders/dumpers (except PhpDumper of course)

Reply:
Oh I didn't see that the container was available in PHPDumper :man_facepalming: